stingray/tests/test_powerspectrum.py

```diff
@@ -53,24 +55,24 @@ def test_periodogram_types(self):
     def test_init_with_lightcurve(self):
         assert Powerspectrum(self.lc)
 
-    @raises(AssertionError)
     def test_init_without_lightcurve(self):
-        assert Powerspectrum(self.lc.counts)
+        with pytest.raises(AssertionError):
+            assert Powerspectrum(self.lc.counts)
 
-    @raises(AssertionError)
     def test_init_with_nonsense_data(self):
         nonsense_data = [None for i in range(100)]
```

Comment:
As to range function - it may be also useful information.
For usage like this (just iterating over list of numbers) better is use `xrange` function.
range - creates a complete list and put it in memory
xrange - creates generator (or some other similar object) and returns next generator state every iteration. so in memory you've got just one value every time not whole list like in range case.

It's just tidbit